### DISCUSSION TITLE
(Poco F3) Update of xDrip+ causes alerts to return to the default alert tone

### DISCUSSION THREAD

**Discussion by dar-kod** [NightscoutFoundation/xDrip#2060] [General]
Version: 524456a-2022.03.27
Phone: Poco F3 (MIUI 12.5.6)

At the end of November 2021, I started to use a new phone, Poco F3. After each update, I have the following issue: all alerts return to the default alert, which is very loud. In the settings, nothing is changed. I have to say that all alerts in my phone are the same. What I do is that I go to Settings -> Alarms and Alerts -> Glucose Level Alerts List -> long click on any created alert and by Alert Tone click at CHOSE FILE -> System Sound/Alarm -> Pre-installed ringtones -> in My ringtones I find my already choosen ringtone and click Apply -> SAVE ALERT. This brings back the already selected Alert Tone. In this order, I will attach screenshots for easier viewing.

![1648413160112](https://user-images.githubusercontent.com/24446806/160299934-36afc89d-5482-44f9-b013-4dd6f172b7c5.jpg)
![1648413160109](https://user-images.githubusercontent.com/24446806/160299938-4fedc37a-5bc3-4dab-a5d0-80eca6d19a6d.jpg)
![1648413160105](https://user-images.githubusercontent.com/24446806/160299943-6a1ccf2b-4b36-4b2c-ad86-0b5f8131f96d.jpg)
![1648413160102](https://user-images.githubusercontent.com/24446806/160299950-71298f34-d6e7-47aa-a0c8-377f34024e21.jpg)

**Comment by Navid200** [comment]
I update xDrip every time there is a new release and I have never experienced this.
So, I am surprised to see this.

@tzachi-dar What could be happening here?

**Comment by tzachi-dar** [comment]
Strange thing, I also update from time to time, and never seen anything alike.
It might be that in the update processes permissions are revoked. if xDrip can not read an alert it goes back to the default sound.
@dar-kod can you check if after update xDrip still has all it's permissions? (especially permission to read from the storage).

**Reply by dar-kod** [reply]
@tzachi-dar Do I need to wait for a new version? If not, I installed now the same latest Nightly version over. In Apps -> Manage apps -> xDrip+ -> App permissions -> File and media -> "Allow management of all files" is ticked. Other allowed permissions are: Location and Physical activity. In Other Permissionas for xDrip+ 5 more permissions are allowed (please see screenshot) and "Modify system settings" is not allowed. Should I allow "Modify system settings"?

![1648461016621](https://user-images.githubusercontent.com/24446806/160373089-91e53246-845a-432b-ac96-6695b3ab247c.jpg)

**Comment by tzachi-dar** [comment]
When there is a new release, please verify if you are still having the same permissions.

**Comment by dar-kod** [comment]
@tzachi-dar, I have tested, there is no need to wait for a new release, just reinstalling the same release will trigger up this problem. The permissions are same before and after managing Alert Tone to sound properly as I described above. Do I need to give xDrip+ permission to modity system settings?

**Reply by Navid200** [reply]
Will you experience the same problem if you use this method?
https://navid200.github.io/xDrip/docs/Custom-Alert-Sound.html

**Reply by dar-kod** [reply]
@Navid200 I placed the mp3 sound file in the Notifications folder, but I do not see it among the existing system sound files.  I tested, this problem doesn't appear, if I use the system sound files as Alert Tone.

**Comment by tzachi-dar** [comment]
you should enable more logging:
Please go to settings->less common settings->extra logging settings->extra tags for logging.
Please enter there AlertPlayer:v
Than have an alert once it works, and then once it does not work, and we will try to see what the difference is.

**Reply by dar-kod** [reply]
These are the logs when it works (alert time: 11:44) and when it doesn't work (alert time: 11:55):

![1648547892836](https://user-images.githubusercontent.com/24446806/160587303-03d65486-76af-44e9-b3a0-a4d4dc313ec7.jpg)
![1648547892838](https://user-images.githubusercontent.com/24446806/160587335-92e110ab-de77-474f-b8de-78b4f79c512e.jpg)

These are the logs when it doesn't work (alert time: 11:55):

![1648547892826](https://user-images.githubusercontent.com/24446806/160587764-a342191f-74e4-4863-804b-7958c9854f22.jpg)
![1648547892829](https://user-images.githubusercontent.com/24446806/160587786-8b749d15-bb11-46f6-b7fc-1ba78fdc2efa.jpg)
![1648547892832](https://user-images.githubusercontent.com/24446806/160587802-733d826b-e1e0-47d4-b80a-5c242989154b.jpg)

**Comment by tzachi-dar** [comment]
The issue is in the line "setmediadata source from uri failed" from 11:55.
I'll try to add more printing to this line later and we can try to gather
more information.

**Comment by tzachi-dar** [comment]
Can you try running with this code:
https://drive.google.com/file/d/1mkbd3DRQRjjn87u4eMT8D4XBhJnfFhNu/view?usp=sharing
you will have to backup xdrip, uninstall it and install this code. This should give more information in the logs.

**Comment by dar-kod** [comment]
Wit backup do you mean "Load / Save Settings"?

**Reply by Navid200** [reply]
No
If you uninstall xDrip, you will lose all your readings history in xDrip

This is how you backup the database:
https://navid200.github.io/xDrip/docs/Backup-Database.html

This is how you restore it:
https://navid200.github.io/xDrip/docs/Restore-Database.html

And, you cannot import a saved database.  You can only replace the existing database with the one you are restoring.

**Reply by dar-kod** [reply]
Thank you. If I am running **xdrip-dev.apk** for some time, do I need to repeat backup and restore the database after installing again nightly version?

**Reply by tzachi-dar** [reply]
Yes, you will have to uninstall and install again, so you will need to back up again.

**Reply by dar-kod** [reply]
Ok, I will do it. It'll need maybe 30 minutes to do all that, but I am not sure, if I can do it today. I hope tomorrow, late in the evening.

**Reply by dar-kod** [reply]
I made backup of the database, uninstalled xDrip+, installed xdrip-dev.apk, repeated to restore database a few times as of Android 11 and imported settings. Now I have BG values for only past 3 hours. Is it ok?

**Reply by dar-kod** [reply]
I imported database many times and each time was coming this screen (where I clicked OK): 

**Import Result**
**_________________________**
Successfully imported 
database
_________________________
OK

**Reply by dar-kod** [reply]
Other thing, my alerts are all disappeared.

**Comment by dar-kod** [comment]
@tzachi-dar ### Logs with xdrip-dev.apk: 
Alert time from 15.44 and above is when it doesn't work, and 15.40 to 15.41 are the logs when it works.

![1648907339079](https://user-images.githubusercontent.com/24446806/161386770-1b52a6af-f5d3-43c8-a631-7dd11918ffdc.jpg)
![1648907339081](https://user-images.githubusercontent.com/24446806/161386782-97206c3c-9554-497c-b74e-e27f9699d567.jpg)

**Comment by Navid200** [comment]
If the database is exported properly and then imported back in, all readings should be present.  The readings from the last 3 hours come from the transmitter, not from the database restore.

Hopefully, you have your readings also stored somewhere else like in Nightscout.
I am not sure what happened with your backup procedure that it is not working.

There are ways to inspect the database you have saved to see how much data in contains.  I personally am not sure how to do it.

**Reply by dar-kod** [reply]
No problem, I have readings in Nightscout. I went back to Nightly and imported database after 3, 4, 5 or 6 times. A few hours of reading are missing, but it is ok. Hope my provided data may help to fix the issue.

**Reply by Navid200** [reply]
I have custom sound files.  They are mp3.  I just place them in the "Notifications" folder and they show up listed with the system sound files.
This has always worked for me for different phones, Android 6, 7, 8, 9, and 11.

**Comment by tzachi-dar** [comment]
@dar-kod I'm afraid I don't see anything obvious in the new code. Will reboot of the phone after the upgrade solve the issue?

**Comment by dar-kod** [comment]
@tzachi-dar sorry I was not clear in the previous writing. I didn't try a reboot, but I uninstalled debug (dev) version and installed nightly version again and tried to import database which I created before uninstalling nightly version. As I have Android 11 I had to repeat import of database many times before data became visible. I lost a few hours of data only in xDrip+, but that is ok.

The problems I had with debug version were also that all alerts diassapeared and in AndroidAPS I got many duplicate BG values, what must be a bug. I reported this in another discussion. 

I hope that alert logs' data from debug version will be helpful for you to fix this issue.
